hub/api/dataset.py

```diff
@@ -227,10 +235,94 @@ def like(
     @staticmethod
     @hub_reporter.record_call
     def ingest(
-        path: str, src: str, src_creds: dict, overwrite: bool = False
+        src: str, dest: str, src_creds: dict, compression: str, overwrite: bool = False
     ) -> Dataset:
-        """Ingests a dataset from a source"""
-        raise NotImplementedError
+        """Ingests a dataset from a source and store it as a structured dataset to destination
+
+        Note:
+            Currently only local source paths and image classification datasets are supported for automatic ingestion.
```

Comment:
We should be more explicit about the assumed underlying data structure in the source. Does it need to have a 1-folder per class structure. What if there are extra folders with random info? What if there are extra files with metadata? What if all the files are in 1 folder and there's another file that maps filenames to classes?

The envelope of what works and what doesn't should be more clear.

Reply:
@istranic I have updated the doctoring to provide more clarity
https://github.com/activeloopai/Hub/blob/25f8666da39630ac69094e1776c1228865e1e619/hub/api/dataset.py#L242-L248